x-pack/plugins/enterprise_search/common/types/app_search.ts

```diff
@@ -30,3 +30,31 @@ export interface IConfiguredLimits {
     maxEnginesPerMetaEngine: number;
   };
 }
+
+export interface IApiToken {
```

Comment:
AFAICT these new type defs are only used in `public/` files, not in `server/` correct? If so, let's go ahead and move them to `public/applications/app_search/types.ts` so that they don't cause a server restart every time they change.

I also know I only just added said file in #77693 so apologies for that extra friction there 🙏

Reply:
75eb3c4dc0670b83aa4dde23e7ebded26ec32911